### PR TITLE
Refactor the charm based on feedback

### PR DESCRIPTION
**Pull request by phvalguima**
Refactor the charm proposal based on the feedback provided by @marcoppenheimer

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -18,110 +18,187 @@
...
+                run_count=self.config.get("run_count", 0),
+                test_mode=self.config.get("test_mode", False),
             )
         )
 
     @override
-    def get_database_options(self) -> DPBenchmarkBaseDatabaseModel:
-        """Returns the database options."""
-        endpoints = self.relation_data.get("endpoints")
+    def render_service_file(
```
Comment (marcoppenheimer) [inline src/charm.py:78]: **question:** Is this duped from what's in `config`?

File: src/charm.py [hunk 2]
```diff
@@ -18,110 +18,187 @@
...
+            config["extra_config"] += " --test_mode"
+        self._render(
+            self.workload.paths.svc_name + ".service.j2",
+            config,
+            dst_filepath=self.workload.paths.service,
+        )
+        return daemon_reload()
+
+
+class OpenSearchDatabaseState(DatabaseState):
```
Comment (marcoppenheimer) [inline src/charm.py:111]: **question:** Is this also duped?

**Review by marcoppenheimer — APPROVED** [review]
Approving, as style wise, this is great.
Would just suggest to do a run-through to check for duped object/methods, I may not have caught them all.

Otherwise great work, this is much tidier! 👍🏾

File: src/charm.py (hunk 1 above)
Comment (phvalguima) [reply src/charm.py:78]: yes, I will merge this PR and remove this left over on the next one.

Comment (phvalguima) [reply src/charm.py:78]: actually, it is not. This is an overload of the main ConfigManager.

File: src/charm.py (hunk 2 above)
Comment (phvalguima) [reply src/charm.py:111]: No, it is a chile class.

**Pull request merged**